### PR TITLE
Allow adding properties to release bundles

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Precondition for: https://github.com/jfrog/jfrog-cli/pull/959, https://github.com/jfrog/jfrog-cli-core/pull/60

Allow adding properties to a created/updated release bundle.
See added_props in https://www.jfrog.com/confluence/display/JFROG/Distribution+REST+API#DistributionRESTAPI-CreateReleaseBundleVersion.

### CODE REVIEW

File: artifactory/services/upload.go [hunk 1]
```diff
@@ -166,7 +169,7 @@ func collectFilesForUpload(uploadParams UploadParams, producer parallel.Runner,
 			}
 			uploadParams.BuildProps += vcsProps
 		}
-		uploadData := UploadData{Artifact: artifact, Props: props, BuildProps: uploadParams.BuildProps}
+		uploadData := UploadData{Artifact: artifact, AddProps: uploadParams.GetAddProps(), BuildProps: uploadParams.BuildProps}
```
Comment (eyalbe4) [inline artifactory/services/upload.go:172]: AddProps --> TargetProps

File: distribution/services/utils/distributionutils.go [hunk 2]
```diff
@@ -31,42 +31,73 @@ func NewReleaseBundleParams(name, version string) ReleaseBundleParams {
...
+		// Create AQL
+		aql, err := createAql(specFile)
+		if err != nil {
+			return nil, err
 		}
-		aql := utils.BuildQueryFromSpecFile(specFile, utils.NONE)
-		bundleQueries = append(bundleQueries, BundleQuery{Aql: aql})
+
+		// Create added properties
+		addProps, err := createAddProps(specFile)
```
Comment (eyalbe4) [inline distribution/services/utils/distributionutils.go:45]: createAddProps --> createTargetProps

File: tests/artifactoryupload_test.go [hunk 3]
```diff
@@ -262,3 +246,53 @@ func explodeUpload(t *testing.T) {
...
+	if failed != 0 {
+		t.Error("Failed to upload", failed, "files.")
+	}
+	if err != nil {
+		t.Error(err)
+	}
+	searchParams := services.NewSearchParams()
+	searchParams.ArtifactoryCommonParams = &utils.ArtifactoryCommonParams{}
+	searchParams.Pattern = RtTargetRepo
+	searchParams.Props = "key1=val2"
```
Comment (eyalbe4) [inline tests/artifactoryupload_test.go:271]: Are you sure this is correct? It looks like the value we're uploading is val1 and not val2, but I may be missing something.
If I am missing it, let's add some comments describing the test flow.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
There are a few AddProps --> TargetProps which I skipped.

File: distribution/services/utils/distributionutils.go (hunk 2 above)
Comment (yahavi) [reply distribution/services/utils/distributionutils.go]: Changed to createAddedProps to be consisted with Distribution REST API

File: tests/artifactoryupload_test.go (hunk 3 above)
Comment (yahavi) [reply tests/artifactoryupload_test.go]: Good catch! That test was broken.

**Pull request merged**